tests/foreman/api/test_content_credentials.py

```diff
@@ -0,0 +1,239 @@
+from fauxfactory import gen_string
+from nailgun import entities
+from requests import HTTPError
+
+from robottelo.constants import VALID_GPG_KEY_BETA_FILE
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
+from robottelo.datafactory import valid_data_list
+from robottelo.helpers import read_data_file
+
+
+key_content = read_data_file(VALID_GPG_KEY_FILE)
+
+
+@pytest.mark.tier1
+def test_positive_search_in_org(module_org):
+    """Search for a GPG key and specify just ``organization_id``.
+
+    :id: ff5e047c-404b-4379-8d28-3ad8cb39b6a9
+
+    :Steps:
+
+        1. Create a GPG key belonging to the organization.
+        2. Search for GPG keys in the organization.
+
+    :expectedresults: Only one GPG key is in the search results: the
+        created GPG key.
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org).create()
+    gpg_keys = gpg_key.search({'organization'})
+    assert len(gpg_keys) == 1
+    assert gpg_key.id == gpg_keys[0].id
+
+
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+@pytest.mark.tier1
+def test_positive_create_with_name(module_org, name):
+    """Create a GPG key with valid name.
+
+    :id: 741d969b-28ef-481f-bcf7-ed4cd920b030
+
+    :parametrized: yes
+
+    :expectedresults: A GPG key is created with the given name.
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org, name=name).create()
+    assert name == gpg_key.name
+
+
+@pytest.mark.tier1
+def test_positive_create_with_content(module_org):
+    """Create a GPG key with valid name and valid gpg key text.
+
+    :id: cfa6690e-fed7-49cf-94f9-fd2deed941c0
+
+    :expectedresults: A GPG key is created with the expected content.
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org, content=key_content).create()
+    assert key_content == gpg_key.content
+
+
+@pytest.mark.parametrize('name', **parametrized(invalid_values_list()))
+@pytest.mark.tier1
+def test_negative_create_name(module_org, name):
+    """Attempt to create GPG key with invalid names only.
+
+    :id: 904a3ed0-7d50-495e-a700-b4f1ae913599
+
+    :parametrized: yes
+
+    :expectedresults: A GPG key is not created and error is raised.
+
+    :CaseImportance: Critical
+    """
+    with pytest.raises(HTTPError):
+        entities.GPGKey(organization=module_org, name=name).create()
+
+
+@pytest.mark.tier1
+def test_negative_create_with_same_name(module_org):
+    """Attempt to create a GPG key providing a name of already existent
+    entity
+
+    :id: 78299f13-5977-4409-9bc7-844e54349926
+
+    :expectedresults: A GPG key is not created and error is raised.
+
+    :CaseImportance: Critical
+    """
+    name = gen_string('alphanumeric')
+    entities.GPGKey(organization=module_org, name=name).create()
+    with pytest.raises(HTTPError):
```

Comment:
If the server wouldn't handle the invalid data correctly, it would return 500 ISE, which I believe this test is supposed to catch. However, the condition in the current form would simply evaluate as true and mark the test Passed.
I think we should be more specific with the status code and possible the message as well.

Reply:
this applies to all `negative` test cases.

Reply:
@rplevka thank you for your review, please take a look at the changes.

Reply:
Re the end-to-end suggestion:
Simplify content credentials test #8330